packages/offchain-resolver-backend/src/utils/intercept.ts

```diff
@@ -0,0 +1,45 @@
+import { log } from 'dm3-lib-shared';
+
+export interface Interceptor {
+    ensName: string;
+    addr?: string;
+    textRecords?: Record<string, string>;
+}
+
+function getInterceptor(ensName: string) {
+    if (process.env.interceptor) {
+        const interceptor: Interceptor = JSON.parse(process.env.interceptor);
+        const higherLevelDomain = ensName.split('.').splice(1).join('.');
+
+        return { interceptor, higherLevelDomain };
+    }
+}
+
+// intercepts a text record query if the higher level domain name is registered
+// in the interceptor env variable
+export function interceptTextRecord(ensName: string, textRecordName: string) {
+    try {
+        const interceptorContainer = getInterceptor(ensName);
```

Comment:
Maybe destructuring the interceptor container might increase readability
`const {interceptor,higherLevelDomain} = getInterceptor(ensName)`

Reply:
I couldn't destruct it since it may return undefined but I moved the `higherLevelDomain`-Check to `getInterceptor` so it should be more readable now